gammapy/modeling/models/spectral.py

```diff
@@ -163,6 +163,26 @@ def integral(self, emin, emax, **kwargs):
         else:
             return integrate_spectrum(self, emin, emax, **kwargs)
 
+    def integral_error(self, emin, emax):
+        """Evaluate the error of the integral flux of a given spectrum in
+        a given energy range.
+
+        Parameters
+        ----------
+        emin, emax : `~numpy.array`
+            Arrays of minimum and maximum energies.
+
+        Returns
+        -------
+        flux, flux_err : tuple of `~astropy.units.Quantity`
+            Integral flux and flux error betwen emin and emax.
+        """
+        dE = emax - emin
+        flux = np.sum(self.integral(emin,emax))
```

Comment:
The idea is, that `flux` as well as `flux_err` can be arrays as well. You can either achieve this by using relative error propagation like `df = f * dg ./ df` or using the approach outlined in the "Flux" section here: https://github.com/gammapy/gammapy-extra/blob/master/experiments/uncertainty_estimation_prototype.ipynb